packages/cli/src/commands/generate/schema-from-database/generate_schema_command.ts

```diff
@@ -67,8 +70,13 @@ export class GenerateSchemaCommand
     const connectionUriSecretName = args.connectionUriSecret as string;
     const outputFile = args.out as string;
 
+    const backendIdentifierIsStack = 'stackName' in backendIdentifier;
+    const sanitizedBackendId = backendIdentifierIsStack
+      ? BackendIdentifierConversions.fromStackName(backendIdentifier.stackName)
+      : backendIdentifier;
+
```

Comment:
Can this be turned into a `convertDeployedBackendIdentifierToBackendIdentifier`? Because we have two types `DeployedBackendIdentifier` and `BackendIdentifier` this confusion or issue may happen else where as well.

Reply:
Or perhaps we should refactor this and have.
`BackendIdentifierResolver.resolveDeployedBackendIdentifier`
and
`BackendIdentifierResolver.resolveBackendIdentifier` 
so that this conversion logic stays in one place.

these are all internal types to cli package so we could do that.

Reply:
I'm leaning towards keeping it all in `BackendIdentifierResolver` for now, we can determine later if there is a need to separate this logic.